### DISCUSSION TITLE
RStudio Cloud Compatibility

### DISCUSSION THREAD

**Discussion by Emask7** [satijalab/seurat#6385] [Q&A]
I am trying to run Seurat to integrate and analyze 8 samples (each with around 10,000 cells each), but RStudio keeps crashing during the integration step. I have tried the advice listed [here](https://satijalab.org/seurat/articles/integration_large_datasets.html), but that still crashes RStudio. I am guessing that the issue is just that my computer doesn't have the processing power needed to integrate that many cells. Does anyone know if this kind of analysis in Seurat is compatible with the RStudio Cloud? I have a feeling that it would not work with the free version, but the paid version gives you 16 GB of RAM and 4 CPUs.

**Comment by samuel-marsh** [comment]
Hi,

Yes it is almost certainly the RAM.  The number of CPUs will help speed things up for things that can be run in parallel but will most likely not be causing failures.

In terms of RStudio cloud directly even 16GB of RAM for larger datasets will probably also be potentially problematic.  However, you can deploy RStudio cloud simply using google cloud platform (lots of blogs and other instructions can find easily with quick google).  That way you can scale the RAM with the size of the datasets on demand (in order to preserve costs).

Best,
Sam

**Reply by Emask7** [reply]
Thank you! I think I found the RStudio google cloud platform you are talking about, but I am not familiar with the google cloud platform. Do you know if there is a way to import an already integrated data set into Seurat? I am thinking about running integration using the aggr pipeline on the 10xGenomics cloud platform, then trying to load the results of that into Seurat if that is possible. That way, I would not have to figure out how to use the Google cloud platform.

**Reply by samuel-marsh** [reply]
So the issue will likely not be solved by running aggr and then importing data.  Aggr’s function is to take multiple sample matrices and combine them into single matrix (and downsample of that option is selected).  It’s not actually going to run integration in the definition of Seurat analysis.  It’s the size of matrix (number of values, columns, rows) is what will determine its size in R so that won’t really save you much memory overall.

There are lots of good resources (including google’s own) that can help you better understand GCP and again lots of other blogs and resources for how to run Rstudio in GCP environment.

Best,
Sam
